hack/verify-gendocs.sh

```diff
@@ -69,6 +70,16 @@ if [[ $ret -gt 1 ]]; then
   exit 1
 fi
 
+"${mungedocs}" "--verify=true" "--root-dir=${TYPEROOT}" "--repo-root=../.." && ret=0 || ret=$?
```

Comment:
I'd really rather split the processing out into separate tools.  It will be net slower but far less brittle.

If we want link-checking for .go files but NOT the rest of the munges, we should make a standalone link-checker tool